docs/spec2006-tutorial.md

```diff
@@ -0,0 +1,587 @@
+We figure out that the dependencies include g++, gcc, and gfortran. 
+So we will get the build-essential and gfortran packages from Debian (note that "12345" is the default password, this could be modified in the spec2006.json file).  
+
+In disk-image/spec2006/install-spec2006.sh,
+```sh
+# install build-essential (gcc and g++ included) and gfortran
+echo "12345" | sudo apt-get install build-essential gfortran
+
+# mount the SPEC2006 ISO file and install SPEC to the disk image
+mkdir /home/gem5/mnt
+mount -o loop -t iso9660 /home/gem5/CPU2006v1.0.1.iso /home/gem5/mnt
+mkdir /home/gem5/spec2006
+echo "y" | /home/gem5/mnt/install.sh -d /home/gem5/spec2006 -u linux-suse101-AMD64
+cd /home/gem5/spec2006
+. /home/gem5/mnt/shrc
+umount /home/gem5/mnt
+rm -f /home/gem5/CPU2006v1.0.1.iso
+
+# use the gcc42 config as the template 
+cp /home/gem5/spec2006/config/linux64-amd64-gcc42.cfg /home/gem5/spec2006/config/myconfig.cfg
+
+# Those 'sed' commands replace paths to gcc, g++ and gfortran binary from /usr/local/sles9/gcc42-0325/bin/* to /usr/bin/*
+sed -i "s/\/usr\/local\/sles9\/gcc42-0325\/bin\/gcc/\/usr\/bin\/gcc/g" /home/gem5/spec2006/config/myconfig.cfg
+sed -i "s/\/usr\/local\/sles9\/gcc42-0325\/bin\/g++/\/usr\/bin\/g++/g" /home/gem5/spec2006/config/myconfig.cfg
+sed -i "s/\/usr\/local\/sles9\/gcc42-0325\/bin\/gfortran/\/usr\/bin\/gfortran/g" /home/gem5/spec2006/config/myconfig.cfg
+
+# build all SPEC workloads
+runspec --config=myconfig.cfg --action=build all
+```
+We don't make Artifact objects for those scripts. 
+Instead, we let the Artifact object of the root folder to keep track of the changes in the above scripts. 
+### Building the Disk Image  
+Having prepared necessary scripts to create the disk image, in this step, we will build the disk image using [packer](https://www.packer.io/).  
+
+First, we download the packer binary.
+```sh
+cd disk-image/
+wget https://releases.hashicorp.com/packer/1.4.5/packer_1.4.5_linux_amd64.zip
+unzip packer_1.4.5_linux_amd64.zip
+```
+In launch_spec_experiments.py, we document how we obtain the binary as follows,  
+```python
+packer = Artifact.registerArtifact(
+    command = '''wget https://releases.hashicorp.com/packer/1.4.5/packer_1.4.5_linux_amd64.zip;
+    unzip packer_1.4.5_linux_amd64.zip;
+    ''',
+    typ = 'binary',
+    name = 'packer',
+    path =  'disk-image/packer',
+    cwd = 'disk-image',
+    documentation = 'Program to build disk images. Downloaded sometime in November from hashicorp.'
+)
+```
+Second, we create a packer script (a json file) that describes how the disk image will be built. 
+In this step, we assume that we have the SPEC 2006 ISO file in the disk-image/spec2006 folder. 
+In this script, the ISO file name is CPU2006v1.0.1.iso. 
+TODO: point to github instead of showing the code. 
+The script, located at disk-image/spec2006/spec2006.json, is as follows,
+```json
```

Comment:
Is there anything in this file that's important to call out? 

(I agree with your "TODO" :))